catalog/video/video-mpegts-stream/src/test/java/org/codice/alliance/video/stream/mpegts/plugins/FindChildrenStreamEndPluginTest.java

```diff
@@ -0,0 +1,150 @@
+        metacard2 = mock(Metacard.class);
+
+        Result result1 = mock(Result.class);
+        when(result1.getMetacard()).thenReturn(metacard1);
+        Result result2 = mock(Result.class);
+        when(result2.getMetacard()).thenReturn(metacard2);
+
+        QueryResponse queryResponse1 = mock(QueryResponse.class);
+        when(queryResponse1.getHits()).thenReturn(2L);
+        when(queryResponse1.getResults()).thenReturn(Collections.singletonList(result1));
+
+        QueryResponse queryResponse2 = mock(QueryResponse.class);
+        when(queryResponse2.getHits()).thenReturn(2L);
+        when(queryResponse2.getResults()).thenReturn(Collections.singletonList(result2));
+
+        when(catalogFramework.query(any())).thenReturn(queryResponse1, queryResponse2);
+
+        UdpStreamProcessor udpStreamProcessor = mock(UdpStreamProcessor.class);
+        when(udpStreamProcessor.getCatalogFramework()).thenReturn(catalogFramework);
+
+        context = new Context(udpStreamProcessor);
+        context.setParentMetacard(parentMetacard);
+
+        handler = mock(FindChildrenStreamEndPlugin.Handler.class);
+
+        FindChildrenStreamEndPlugin.Factory factory =
+                mock(FindChildrenStreamEndPlugin.Factory.class);
+
+        when(factory.build()).thenReturn(handler);
+
+        findChildrenStreamEndPlugin = new FindChildrenStreamEndPlugin(new GeotoolsFilterBuilder(),
+                factory);
+
+    }
+
+    @Test
+    public void testBatching()
+            throws UnsupportedQueryException, SourceUnavailableException, FederationException {
+
+        setIsParentDirty(true);
+
+        findChildrenStreamEndPlugin.streamEnded(context);
+
+        ArgumentCaptor<List> argumentCaptor = ArgumentCaptor.forClass(List.class);
+
+        verify(handler, times(2)).handle(Matchers.eq(context),
+                Matchers.eq(parentMetacard),
+                argumentCaptor.capture());
+        verify(handler).end(Matchers.eq(context), Matchers.eq(parentMetacard));
+
+        assertThat(argumentCaptor.getAllValues()
+                .get(0)
+                .get(0), is(metacard1));
+        assertThat(argumentCaptor.getAllValues()
+                .get(1)
+                .get(0), is(metacard2));
+        assertThat(context.modifyParentOrChild(AtomicBoolean::get), is(false));
+
+    }
+
+    /**
+     * Test that when the parent metcard is not dirty, that the catalog framework does not get called.
+     */
+    @Test
+    public void testIsParentDirty()
+            throws UnsupportedQueryException, SourceUnavailableException, FederationException {
+
+        setIsParentDirty(false);
+
+        findChildrenStreamEndPlugin.streamEnded(context);
+
+        verify(catalogFramework, times(0)).query(Matchers.any());
+
+    }
+
+    private void setIsParentDirty(boolean value) {
+        context.modifyParentOrChild(isParentDirty -> {
+            isParentDirty.set(value);
+            return (Void) null;
+        });
+    }
+
```

Comment:
Not necessarily. If we expect `RuntimeException`s to just bubble up and stop the retries, then we should have a test that proves it's the case.